tests/formats/transition/README.md

```diff
@@ -0,0 +1,72 @@
+# Transition testing
+
+Transition tests to cover processing the chain across a fork boundary.
+
+Each test case contains a `post_fork` key in the `meta.yaml` that indicates the target fork which also fixes the fork the test begins in.
+
+Clients should assume forks happen sequentially in the following manner:
+
+0. `phase0`
+1. `altair`
+
+For example, if a test case has `post_fork` of `altair`, the test consumer should assume the test begins in `phase0` and use that specification to process the initial state and any blocks up until the fork epoch. After the fork happens, the test consumer should use the specification according to the `altair` fork to process the remaining data.
+
+## Encoding notes
+
+This test type contains objects that span fork boundaries.
+In general, it may not be clear which objects belong to which fork so each
+object is prefixed with a SSZ `boolean` to indicate if the object belongs to the post fork or if it belongs to the initial fork.
+This "flagged" data should be used to select the appropriate version of the spec when interpreting the enclosed object.
+
+```python
+class FlaggedContainer(Container):
+    flag: boolean
+    obj: Container
+```
+
+If `flag` is `False`, then the `obj` belongs to the **initial** fork.
+If `flag` is `True`, then the `obj` belongs to the **post** fork.
```

Comment:
Given that only the blocks are prefixed, I'd be tempted to simplify this and just record in the metadata file the number of first post-fork block.  Not the slot number, but more directly recording that the first 5 blocks are pre-fork and the 6th block onwards are all post-fork.  That way the test runner can decide ahead of time how to decode the file rather than having to parse the initial boolean and then switching to the appropriate block schema.

That said, in Teku we already have code to extract the slot from blocks without knowing what fork it's from so at least for us we wouldn't have any issues decoding just the raw blocks given we know the fork_epoch. We could probably work with this `FlaggedContainer` by just trimming the prefix it adds and then doing our usual fork-agnostic block parsing.

Reply:
yeah that is not a bad suggestion and simplifies the test format... main tension is how general we want this type of test to be -- can we think of some fork condition where a "fork index" would bind our hands? i can't think of any at the moment....

and the thing w/ taking the slot from the raw block bytes is that the position of the `slot` could change, e.g. in a future version of `ssz`. so i'd rather stay at a higher level of abstraction than needing to peer into "opaque" `BeaconBlock` bytes...